packages/drivers/odsp-driver/src/odspDocumentStorageServiceBase.ts

```diff
@@ -208,6 +210,13 @@ export abstract class OdspDocumentStorageServiceBase implements IDocumentStorage
 		return this.combineProtocolAndAppSnapshotTree(appTree, protocolTree);
 	}
 
+	public async getSnapshot(
+		version?: api.IVersion,
+		snapshotFetchOptions?: ISnapshotFetchOptions,
+	): Promise<ISnapshot | undefined> {
+		throw new UsageError("Not implemented yet!");
+	}
```

Comment:
Can we avoid having implementation like that?
Is using abstract function sufficient here (i.e. we are satisfying interface, but push implementation to all derived classes).
Having NYI function means - someone will get a suprise. They will test for presence of this function on interface, and assume it's there, meaning new contract is implemented, when it's not.

Reply:
Derived classes need to implement abstract functions. So, added implementation where required. The implementation is not complete yet for the api so added assert to make sure this is not called yet.